scripts/illumina_run_parameter_parser.py

```diff
@@ -0,0 +1,233 @@
+    process.udf['Status'] = "Cycle {} of {}".format(completed_cycles, planned_cycles)
+    process.udf['Flow Cell ID'] = runParameters['FlowCellSerialNumber']
+    process.udf['Experiment Name'] = runParameters['FlowCellSerialNumber']
+    process.udf['Read 1 Cycles'] = int(runParameters['PlannedCycles']['Read1'])
+    process.udf['Index 1 Read Cycles'] = int(runParameters['PlannedCycles']['Index1'])
+    process.udf['Index 2 Read Cycles'] = int(runParameters['PlannedCycles']['Index2'])
+    process.udf['Read 2 Cycles'] = int(runParameters['PlannedCycles']['Read2'])
+    process.udf['Run ID'] = runParserObj.runinfo.data['Id']
+    process.udf['Reagent Cartridge ID'] = runParameters['CartridgeSerialNumber']
+    # Put in LIMS
+    process.put()
+
+
+def lims_for_miseq(process, run_dir):
+    # Parse run
+    runParserObj, RunParametersParserObj = parse_run(run_dir)
+    # Set values for LIMS UDFs
+    runParameters = RunParametersParserObj.data['RunParameters']
+    process.udf['Finish Date'] = datetime.now().date()
+    if runParameters['Setup']['SupportMultipleSurfacesInUI'] == 'true' and runParameters['Setup']['NumTilesPerSwath'] == '19':
+        process.udf['Run Type'] = 'Version3'
+    elif runParameters['Setup']['SupportMultipleSurfacesInUI'] == 'true' and runParameters['Setup']['NumTilesPerSwath'] == '14':
+        process.udf['Run Type'] = 'Version2'
+    elif runParameters['Setup']['SupportMultipleSurfacesInUI'] == 'false' and runParameters['Setup']['NumTilesPerSwath'] == '2':
+        process.udf['Run Type'] = 'Version2Nano'
+    else:
+        process.udf['Run Type'] = 'null'
+    total_cycles = sum(list(map(int, [read['NumCycles'] for read in runParameters['Reads']['RunInfoRead']])))
+    process.udf['Status'] = "Cycle {} of {}".format(total_cycles, total_cycles)
+    process.udf['Flow Cell ID'] = runParameters['FlowcellRFIDTag']['SerialNumber']
+    process.udf['Flow Cell Version'] = runParameters['FlowcellRFIDTag']['PartNumber']
+    process.udf['Experiment Name'] = process.all_inputs()[0].name
+
+    non_index_read_idx = [read['Number'] for read in runParameters['Reads']['RunInfoRead'] if read['IsIndexedRead'] == 'N']
+    index_read_idx = [read['Number'] for read in runParameters['Reads']['RunInfoRead'] if read['IsIndexedRead'] == 'Y']
+    if len(non_index_read_idx) == 2:
+        process.udf['Read 1 Cycles'] = int(list(filter(lambda read: read['Number'] == str(min(list(map(int,non_index_read_idx)))), runParameters['Reads']['RunInfoRead']))[0]['NumCycles'])
+        process.udf['Read 2 Cycles'] = int(list(filter(lambda read: read['Number'] == str(max(list(map(int,non_index_read_idx)))), runParameters['Reads']['RunInfoRead']))[0]['NumCycles'])
+    elif len(non_index_read_idx) == 1:
+        process.udf['Read 1 Cycles'] = int(list(filter(lambda read: read['Number'] == str(min(list(map(int,non_index_read_idx)))), runParameters['Reads']['RunInfoRead']))[0]['NumCycles'])
```

Comment:
Since Read 1 cycles will always be populated, it can't it be outside the `if..elif` ? So you could have something similar to the below to not repeat the commands.
```
if len(non_index_read_idx) >0:
       process.udf['Read 1 Cycles'] = ...
       if len(non_index_read_idx) >1:
               process.udf['Read 2 Cycles'] = ...
```

Reply:
Yes, I will fix it